src/Jobs/TileSingleOverlay.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Biigle\Modules\Geo\Jobs;
+
+use FileCache;
+use Biigle\Modules\Geo\GeoOverlay;
+use Exception;
+use File;
+use FilesystemIterator;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Storage;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use VipsImage;
+
+class TileSingleOverlay extends Job implements ShouldQueue
+{
+    use InteractsWithQueue, SerializesModels;
+
+    /**
+     * The overlay to generate tiles for.
+     *
+     * @var GeoOverlay
+     */
+    public $overlay;
+
+    /**
+     * Path to the temporary storage file for the tiles.
+     *
+     * @var string
+     */
+    public $tempPath;
+
+    public $file;
+
+    /**
+     * Ignore this job if the overlay does not exist any more.
+     *
+     * @var bool
+     */
+    protected $deleteWhenMissingModels = true;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param GeoOverlay $overlay The Overlay to generate tiles for.
+     *
+     * @return void
+     */
+    public function __construct(GeoOverlay $overlay)
+    {
+        $this->overlay = $overlay;
+        $this->file = new File($overlay->file);
+        $this->tempPath = config('overlay.tiles.tmp_dir')."/{$overlay->id}";
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        try {
+            FileCache::getOnce($this->file, [$this, 'generateTiles']);
+            $this->uploadToStorage();
+            $this->overlay->tilingInProgress = false;
+            $this->overlay->save();
+        } finally {
+            File::deleteDirectory($this->tempPath);
+        }
+    }
+
+    /**
+     * Generate tiles for the image and put them to temporary storage.
+     *
+     * @param string $path Path to the cached image file.
+     */
```

Comment:
@mzur, I have trouble understanding the generateTiles() function, and how it can be adapted to be used with the geotiff file. The call to FileCache::getOnce() in the handle() function above results in an error, as it expects a file of type "Biigle\FileCache\Contracts\File". I'm not sure why, since the [FileCache-Interface](https://github.com/biigle/laravel-file-cache/blob/master/src/Contracts/FileCache.php) specifies the first argument to be of type "File". Any help would be highly appreciated!

Reply:
Thanks a lot, this solved the issue!